### ISSUE TITLE
subject rewriting responses are handled incorrectly

### ISSUE THREAD

**Issue by aricart** [nats-io/nats.go#555]
https://github.com/nats-io/nats.go/blob/master/nats.go#L2652

If there's a response mux, services that re-write responses won't be able to parse the response subject to dispatch to the mux subscription. Currently, if there's only one mux subscription it works, however, this means that a client can only make a single request. The option needs to be per-request to accommodate interactions with multiple services at the same time.

Currently, nats.js does have a global option to disable mux requests (making the client create a subscription to handle the request), but this option really ought to be at the request level as well. The all/nothing approach is not flexible enough.

**Comment by derekcollison** [comment]
Yep, when we do syntactic sugar for pulling next messages we will not use the multiplexed _INBOX for general requests.
